src/data/device_registry.ts

```diff
@@ -20,6 +22,33 @@ export interface DeviceRegistryEntryMutableParams {
   name_by_user?: string | null;
 }
 
+export const computeDeviceName = (
+  device: DeviceRegistryEntry,
+  hass: HomeAssistant,
```

Comment:
By having `hass` be passed in, it becomes difficult to memoize.

Reply:
We only use `hass` for `localize` and `fallback`, no need to memoize that I think?